File: October/14th October,2024/Leetcode(2530).cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<long long>pq;
        for(int i:nums) pq.push(i);
        long long ans=0;
        while(k-->0){
            double x=pq.top();
            ans+=x;
            pq.pop();
            pq.push(ceil(x/3));
        }
        return ans;
    }
};

File: October/14th October,2024/Leetcode(2530).java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer>pq=new PriorityQueue<>((a,b)->(b-a));
        for(int i:nums) pq.add(i);
        long ans=0;
        while(k-->0){
            int x=pq.peek();
            ans+=x;
            pq.poll();
            pq.add((int)Math.ceil(x/3.0));
        }
        return ans;
    }
}